pynxtools/dataconverter/readers/xps/xml/xml_specs.py

```diff
@@ -0,0 +1,751 @@
+"""
+Parser for reading XPS (X-ray Photoelectron Spectroscopy) data from
+Specs Lab Prodigy XML export, to be passed to mpes nxdl
+(NeXus Definition Language) template.
+"""
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import xml.etree.ElementTree as EmtT
+from typing import Tuple, List, Any
+import copy
+import xarray as xr
+import numpy as np
+
+from pynxtools.dataconverter.readers.xps.reader_utils import (
+    XPSMapper,
+    construct_entry_name,
+)
+
+
+def _construct_entry_name_xml(key):
+    """Construction entry name."""
+    key_parts = key.split("/")
+    try:
+        # entry example : vendor__sample__name_of_scan_region
+        entry_name = (
+            f"{key_parts[2]}"
+            f"__"
+            f'{key_parts[3].split("_", 1)[1]}'
+            f"__"
+            f'{key_parts[5].split("_", 1)[1]}'
+        )
+    except IndexError:
+        entry_name = ""
+    return entry_name
+
+
+class XmlMapperSpecs(XPSMapper):
+    """
+    Class for restructuring xml data file from
+    specs vendor into python dictionary.
+    """
+
+    config_file = "config_xml_specs.json"
+
+    def _select_parser(self):
+        """
+        Select Specs XML parser.
+        Currently, there is only one parser.
+        Returns
+        -------
+        XmlParserSpecs
+            Parser for reading .xml file exported by SpecsLab2.
+
+        """
+        return XmlParserSpecs()
+
+    def parse_file(self, file, **kwargs):
+        """
+        Parse the file using the parser that fits the Prodigy SLE version.
+        Returns flat list of dictionaries containing one spectrum each.
+
+        """
+        self.file = file
+        self.parser = self._select_parser()
+        metadata_dict, self.raw_data = self.parser.parse_file(file, **kwargs)
+
+        self._xps_dict = {**self._xps_dict, **metadata_dict}
+
+        file_key = f"{self._root_path}/File"
+        self._xps_dict[file_key] = file
+
+        self.construct_data()
+
+        return self.data_dict
+
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-statements
+    # pylint: disable=too-many-branches
```

Comment:
you can use all pylints in one line:
such as 
# pylint: disable=too-many-locals,too-many-statements,...

Reply:
Fixed across all files, thanks